.travis.yml

```diff
@@ -4,17 +4,35 @@ env:
   - secure: oC18EAj4W0vbA9Z9wgS8cXnp4EgGa91xsNkFx4BTc+2zY3UOk6qr+AVRmm09phpStOyNxOijsSloDKhfuB6fQRlv3XDoEIqhrKrA16M6oUltkOJOfa+dxVaeHf4Jd0Pu4Z5QFrFl6gx4oJCAsmJ/vnwoF7+LbM1YNVBz0A4HAut/JVDFqkLACRPJw82F10a03XT6p0yephWqzxCCCsqrQJOCIDBzeiG6WGnPlb++RFQUe6+9GlTAWpm9rOWOEUHoi13mcJyaJn1XUUWTgE0jXYXUMnx8/zQaL5RyEdd9gzOo4B/TR0IoWf4PylkG3936aZJFrg3PRkPs5NxOkFupKOqGGL2pCaN4J0SjIF+z0fYAyqij4NhF/Gr6uWmuxbcllGrhq5Ua2JK3QWJKWm/jt/l3Ulw1Yx4Dt3H5P4+5X+b2saJiEXm18HYPL3MbBDGTe6zNOX2CxapAX2G6hKKSXcHm1azHtl4m6xZLs1G0dBUjEPstF6uZWZ2C5CaLfnxridNm7bLci6k6nr40ARmY/BCbJN/OMnO9HhsgMwYXO3iwtOOGJJPAdPRu6K0+zNvk4to79sLZSJuJrHNV1YwvRbAt92zmAUExHC12iqL4mh9bQmLbXn9yhsh9jBbcxYbKmhFJTX+hqfZE9drErJ9cKHZJw7qnn7FKLJKiq375k7U=
   - secure: I9YCyTjhIkracIXLp29l+hyoGa1cJnVaFELVtG0guv+GpUpuUIxVBMbR7v4EXElX6W/ihGgOf1xKZVdwDSpBPqsn2qKZhTBJkNIfmPqKavlWUcfHT+HBeVybt4QqkmumuoyXLZOfyfvc/M1otlhdSpiJ3kqTp9jeBDoyxT4nwRLnw8dtF4r4gRaolr3TfRrOkndAMuV8QttMzT2g91YoNBVlB8e9WzCTpY+ilnjbgUR73qNJVuT3P88vsh/8vPeTtB2QAum8KK//kcjoKwW75MoT8osR5g3ahU/0OdbGDJsSuCgGyo7Wil1BoiHvpswZplTbxuyq5xFULZhkOelnFF3Be43B4DOO+zQMD8zAAavUO2ZUXO6j6qlmoJNJx1dtT7ZeEwNv3pQSNfcVpwd8olgKeR0NLFd67xgwDly2uCNL7o7/S1Kcc0B0FdaOo2AoxUU/+cldThXhyt4ceoqIe9LtS/xyFCwAJW1d+MFlg1jBCQ2/Jhrh/YNCAsDqndDIUMruecLDVNrZvx9pBEiOBE3aoIK5Qa+lZXTv+o61i2NUeC/g9b8wE44rEKoxRrI+rNVI8JfNaKcD9/10eUQqjcx+dlL31TCtwZydUid2DpjHh4c6QgYxAYTRQeCen2WSi/NOnJXFEoBlluWIOhfqWuihkk0dwf7ZXevueiLX9Q0=
+  - DOCKER_USER=khalidashraf
+  - secure: WrccS5O4eSH8/HKUukA+cJxa+2wCkj/f4lYgWBeGFpoX2xFVICYiYfp5cHSL6VYFCieq24LCufdhB7JvJ+JWjbmmrElC4aluNSfRgpwQBlN7OTVmMwCEzII3Uu8tIww8md7/mIdG/6YBSDPWFPo17tFI69bx0t1AXUWts3WLHTuVilp3EFrlZXpnUKR3d91hSQ+9hG3UnGP8fLAlC/D2b/3gVuCZ7tv9zrcQWh4zFDYIjxT6SKeiOOwwlC9i9qGb6pcDxFxHjMVgXtLU1ue6Gh9KPXyd3WRyPmKd/mxrH8qCw6tad+/gsSk25Vkue+dKQs8F4+Qkc8L2IwcdwIq8ty12Wfu2Wc7GSC7mSCdwq1AySRqOVcAWIlCSGPHP7Rv3bC7RQ0i1CQaRlV4ch/3LfXBva1DfFG01gAKqimmwvT0yCxN8g+MT7pnYfSofnCFFZsuCpahyT1B36wkB9hBAxeUYRO1vi1+scu48J++YfXBqXyvgZm+CnsHv1UsWAYdt5YT1VRnOumJ564wZS64x4MUyKqsFDZvdpB6VICmtwJYk2fcQ4UQ1jL6Hvkezqh7BJew1Osw3LSRr3Y/90iywMAF70/bEKS0E3s2xZVVfblxDOEtlBoEFIgNyBR/TX2fQ2Vi2xo9i0gtV8JtFmwUaYnKzmEsckGaTSyruuYAgbjs=
 services:
 - docker
 script:
+# Build test image and test the code
 - docker build --target test --tag todo_app:test .
 - docker run --env-file .env.test todo_app:test tests
 - docker run -e SECRET_KEY -e TRELLO_KEY -e TRELLO_TOKEN -e BOARD_ID todo_app:test tests_e2e
+
+# Build prod Docker image and push to Docker repo
+- echo $DOCKER_PASS | docker login --username $DOCKER_USER --password-stdin
+- docker build --target prod --tag $DOCKER_USER/todo_app:latest --tag $DOCKER_USER/todo_app:$TRAVIS_COMMIT .
+- docker push $DOCKER_USER/todo_app
+
+# Pull latest prod docker image from Docker repo and push to Heroku repo
+- docker pull $DOCKER_USER/todo_app
+- docker tag $DOCKER_USER/todo_app registry.heroku.com/$DOCKER_USER-todo-app/web
```

Comment:
There's no reason for your Heroku app name to be tied to your Docker username. I'd probably decouple these and have a second env var $HEROKU_APP